pkg/proxy/internal/util_test.go

```diff
@@ -35,3 +37,47 @@ func TestGetDefaultProviderFile(t *testing.T) {
 		})
 	}
 }
+
+func TestWriteFile(t *testing.T) {
```

Comment:
I'm curious: Where you able to reproduce the issue with this test?

Reply:
I'm happy to say YES!
Here is the output of this test with the previous implementation:
```
--- FAIL: TestWriteFile (1.09s)
    --- FAIL: TestWriteFile/ValidData (1.09s)
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
        util_test.go:84: 
            Expected
                <string>: 
            to equal
                <string>: key: value\n
FAIL
FAIL	github.com/canonical/microk8s-cluster-agent/pkg/proxy/internal	1.095s
FAIL
```

Reply:
To be more specific: `5370 out of 10000 times` (using atomic.Int32)

Reply:
nice! Great find @HomayoonAlimohammadi